_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,644 @@
+                 exports.onExecutePostLogin = async (event, api) => {                                                     
+                   const namespace = 'https://www.jhipster.tech';                                                         
+                   if (event.authorization) {                                                                             
+                     api.idToken.setCustomClaim('preferred_username', event.user.email);                                  
+                     api.idToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);                         
+                     api.accessToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);                     
+                   }                                                                                                      
+                 };                                                                                                       
+
+
+                 /**                                                                                                      
+                  * Handler that will be invoked when this action is resuming after an external redirect. If your         
+                  * onExecutePostLogin function does not perform a redirect, this function can be safely ignored.         
+                  *                                                                                                       
+                  * @param {Event} event - Details about the user and the context in which they are logging in.           
+                  * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.  
+                  */                                                                                                      
+                 // exports.onContinuePostLogin = async (event, api) => {                                                 
+                 // };                                                                                                    
+
+```
+
+You can list the available actions with the following command:
+
+```shell
+auth0 actions list
+```
+The output will show the deployment status of each action:
+
+```text
+=== dev-avup2laz.us.auth0.com actions
+
+  ID                                    NAME      TYPE        STATUS  DEPLOYED  
+  ee0ea308-0e50-4be3-893b-74f9ccbb3703  jhipster  post-login  built   ✗         
+```
+
+Note the `DEPLOYED` status is `x`. Go ahead and deploy it using the action ID:
+
+```shell
+auth0 actions deploy ee0ea308-0e50-4be3-893b-74f9ccbb3703
+```
+
+Once the action is deployed, you must attach it to the login flow. Sign in to Auth0, and on the left menu choose **Actions**. Then in the Flows screen, choose **Login**. On the right bottom section, choose **Custom**. Drag the **jhipster** action to the flow diagram. Then click **Apply**. The flow should look like the following:
```

Comment:
Are you sure the dashboard has to be used here and can't be added with the CLI?

Reply:
I could not find an auth0 command for flow management. Please let me know if you found some related doc.

Reply:
i'll check with our tools team

Reply:
Seems like there is no direct command but can be done using the[ `api` command](https://auth0.github.io/auth0-cli/auth0_api.html) calling the [Update trigger bindings](https://auth0.com/docs/api/management/v2#!/Actions/patch_bindings)

Reply:
Added api command to the post.